monai/apps/auto3dseg/bundle_gen.py

```diff
@@ -309,20 +364,40 @@ class BundleGen(AlgoGen):
         python -m monai.apps.auto3dseg BundleGen generate --data_stats_filename="../algorithms/data_stats.yaml"
     """
 
-    def __init__(self, algo_path: str = ".", algos=None, data_stats_filename=None, data_src_cfg_name=None):
+    def __init__(
+        self,
+        algo_path: str = ".",
+        algos: Optional[dict] = None,
+        algo_retrival_methods: Union[Sequence[AlgoRetrieval], AlgoRetrieval] = AlgoRetrieval.REL,
+        algo_urls_or_dirs: Optional[Union[Sequence[str], str]] = AlgoRetrieval.REPO_URL,
+        data_stats_filename=None,
+        data_src_cfg_name=None,
+    ):
         self.algos: Any = []
 
-        if algos is None or isinstance(algos, str):
-            # trigger the download process
-            zip_download_dir = TemporaryDirectory()
-            algo_compressed_file = os.path.join(zip_download_dir.name, "algo_templates.tar.gz")
-            download_and_extract(default_algo_zip if algos is None else algos, algo_compressed_file, algo_path)
-            zip_download_dir.cleanup()
-            sys.path.insert(0, os.path.join(algo_path, "algorithm_templates"))
-            algos = deepcopy(default_algos)
-            for name in algos:
-                algos[name]["template_path"] = os.path.join(
-                    algo_path, "algorithm_templates", algos[name]["template_path"]
+        if not algos:
+            # trigger the download or copy process
+            algos = {}
+            template_path = os.path.join(algo_path, "algorithm_templates")
+            for (method, url_or_dir) in zip(ensure_tuple(algo_retrival_methods), ensure_tuple(algo_urls_or_dirs)):
+                if method is AlgoRetrieval.REL:
+                    _download_release_templates(RELEASE_ALGO, template_path)
+                elif method is AlgoRetrieval.DEV:
+                    _download_dev_templates(url_or_dir, template_path)
+                elif method is AlgoRetrieval.LOCAL:
+                    _copy_algorithm_templates(url_or_dir, template_path)
```

Comment:
I think you can simplify it a lot. 

- Remove `AlgoRetrieval class, and algo_retrival_methods options`
- simply `use algo_urls_or_dirs`. If it's URL then use it to download (the defaults to REPO_URL), but it could be a user URL, If it's local folder, then use a local folder
- `template_path = os.path.join(algo_path, "algorithm_templates")` append hardcodded string algorithm_templates only if downloading from url
- if it's a local folder , there is no need to call _copy_algorithm_templates.  Simple set `template_path = url_or_dir`
- sys.path insert can be moved inside the "if dictionary ".
- capitalizing first letter + str(Algo) is very hack-ish. We should define a standard, e.g. algo file should have class Algo().  This can be achieved if rename all 4 classes, or simply add at the bottom Algo = DintsAlgo in each file